lib/matplotlib/backend_bases.py

```diff
@@ -3165,20 +3165,28 @@ def press_zoom(self, event):
             self.push_current()  # set the home button to this view
         id_zoom = self.canvas.mpl_connect(
             "motion_notify_event", self.drag_zoom)
+        # A colorbar is one-dimensional, so we extend the zoom rectangle out
+        # to the edge of the axes bbox in the other dimension. To do that we
+        # store the orientation of the colorbar for later.
+        if hasattr(axes[0], "_colorbar"):
+            cbar = axes[0]._colorbar.orientation
+        else:
+            cbar = None
         self._zoom_info = self._ZoomInfo(
             direction="in" if event.button == 1 else "out",
-            start_xy=(event.x, event.y), axes=axes, cid=id_zoom)
+            start_xy=(event.x, event.y), axes=axes, cid=id_zoom, cbar=cbar)
 
     def drag_zoom(self, event):
         """Callback for dragging in zoom mode."""
         start_xy = self._zoom_info.start_xy
         ax = self._zoom_info.axes[0]
         (x1, y1), (x2, y2) = np.clip(
             [start_xy, [event.x, event.y]], ax.bbox.min, ax.bbox.max)
-        if event.key == "x":
+        if event.key == "x" or self._zoom_info.cbar == "horizontal":
```

Comment:
Currently one can press `x` on a vertical colorbar and get a nonsensical rubberband (which does nothing on release, but still...)  I guess you could instead do something like
```
key = event.key
if self._zoom_info.cbar == "horizontal": key = "x"  # force it.
if self._zoom_info.cbar == "vertical": key = "y"
if key == "x": ...
if key == "y": ...
```

Reply:
Good catch, updated.